ansible_collections/arista/avd/roles/eos_designs/python_modules/underlay/ip_access_list.py

```diff
@@ -0,0 +1,184 @@
+            ip_access_lists.extend(acl_inbound_lists)
+        if acl_outbound := self._get_access_list_outbound():
+            ip_access_lists.append(acl_outbound)
+
+        if ip_access_lists:
+            return ip_access_lists
+
+        return None
+
+    def _get_access_list_outbound(self) -> dict | None:
+        """
+        Returns the outbound access list.
+        Blocks the RFC1918 traffic.
+        """
+        if self.shared_utils.wan_interfaces:
+            outbound_access_list = {
+                "name": "WAN_TRANSPORT_OUT",
+                "entries": [
+                    {"sequence": self._get_sequence(True), "remark": " ### Blocking RFC 1918 prefixes ###"},
+                    {"sequence": self._get_sequence(), "action": "deny", "protocol": "ip", "source": "any", "destination": "10.0.0.0/8"},
+                    {"sequence": self._get_sequence(), "action": "deny", "protocol": "ip", "source": "any", "destination": "172.16.0.0/12"},
+                    {"sequence": self._get_sequence(), "action": "deny", "protocol": "ip", "source": "any", "destination": "192.168.0.0/16"},
+                    {"sequence": self._get_sequence(), "action": "permit", "protocol": "ip", "source": "any", "destination": "any"},
+                ],
+            }
+            return outbound_access_list
+        return None
+
+    def _get_access_lists_inbound(self) -> List | None:
+        """
+        Returns the inbound access lists for different interfaces.
+        The ACL name per interface is of the form "WAN_TRANSPORT_<INTERFACE_NAME>_IN".
+        """
+        inbound_access_lists = []
+        for l3_interface in self.shared_utils.wan_interfaces:
+            interface_ip = l3_interface.get("ip_address")
+            # Need to see how to handle dhcp cases.
+            if interface_ip and interface_ip != "dhcp":
+                subnet = str(ip_network(interface_ip, False))
+                l3_interface_name = l3_interface.get("name")
+                inbound_access_list = {
+                    "name": self._get_inbound_wan_acl_name(l3_interface_name),
+                    "entries": [],
+                }
+                # Allow BGP
+                inbound_access_list["entries"].extend(
+                    [
+                        {
+                            "sequence": self._get_sequence(refresh=True),
+                            "remark": "### Allow BGP ###",
+                        },
+                        {
+                            "sequence": self._get_sequence(),
+                            "action": "permit",
+                            "protocol": "tcp",
+                            "source": "any",
+                            "destination": subnet,
+                            "destination_ports_match": "eq",
+                            "destination_ports": ["bgp"],
+                        },
+                    ]
+                )
+
+                # Allow GRE
```

Comment:
We don't have needs for GRE anywhere in the supported features of AVD. If we need it later, we can add it to the ACL.